cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/TranslatedCall.qll

```diff
@@ -424,12 +425,15 @@ class TranslatedCallSideEffects extends TranslatedSideEffects, TTranslatedCallSi
 }
 
 class TranslatedStructorCallSideEffects extends TranslatedCallSideEffects {
-  TranslatedStructorCallSideEffects() { getParent().(TranslatedStructorCall).hasQualifier() }
+  TranslatedStructorCallSideEffects() {
+    getParent().(TranslatedStructorCall).hasQualifier() and
```

Comment:
```suggestion
    getParent() instanceof TranslatedStructorCall and
```
Can we actually have `TranslatedStructorCall`s without qualifiers? This seems like something that was originally there to cover for extractor issues with constructor/destructor ASTs, but we've done a lot of work on that since then.

Reply:
I think there's still some funny business in constructor initializer lists where we don't always have a qualified in the AST.